src/controllers/payment.controller.js

```diff
@@ -0,0 +1,173 @@
+import Stripe from 'stripe';
+import { CartService } from '../services/cart.service';
+import { OrderService } from '../services/order.service';
+import { PaymentService } from '../services/payment.service';
+import { SalesService } from '../services/sales.service';
+import { calculateCartTotal } from '../utils/cart.util';
+const stripe = new Stripe(process.env.STRIPE_SECRET_KEY);
+const endpointSecret = process.env.STRIPE_WEBHOOK_ENDPOINT_SECRET;
+
+export const stripeCheckoutSession = async (req, res) => {
+	try {
+		const cart = req.cart;
+		const user = req.user;
+		const { products } = cart;
+		const ids = products.map((product) => product.productId);
+		const productsData = await CartService.getCartProducts(ids);
+		const productsDataWithQuantity = productsData.map((product, index) => ({
+			name: product.name,
+			price: product.price,
+			quantity: products[index].quantity,
+			images: product.images,
+		}));
+		const cartTotal = calculateCartTotal(productsDataWithQuantity);
+		const lineItems = productsData.map((product, index) => ({
+			price_data: {
+				currency: 'rwf',
+				unit_amount: product.price,
+				product_data: {
+					name: product.name,
```

Comment:
Let's refactor this controller by splitting it into different methods and middleware if they are